validator/sawtooth_validator/state/state_delta_processor.py

```diff
@@ -119,6 +120,37 @@ def add_subscriber(
         LOGGER.debug('Added Subscriber %s for %s',
                      connection_id, address_prefix_filters)
 
+        if known_block_id != self._block_store.chain_head.identifier:
+            LOGGER.debug('Catching up Subscriber %s from %s to %s',
+                         connection_id,
+                         known_block_id,
+                         self._block_store.chain_head.identifier[:8])
+            catch_up_blocks = []
+
+            try:
+                for block in self._block_store.get_predecessor_iter():
```

Comment:
The reason I didn't use that, is I wasn't sure that using the predecessor iterator as the default iterator was necessarily the right path.   I wanted this use to be explicit, in case that changes.